### ISSUE TITLE
mypy does not recognize __qualname__

### ISSUE THREAD

**Issue by jtmoon79** [python/mypy#6473]
1. Given Python file `test1.py`
```Python
class A(object):
    NAME = __qualname__
```
2. run mypy checking

       mypy test1.py

3. **_BUG:_** mypy reports `not defined`

       test1.py:2: error: Name '__qualname__' is not defined

The Python interpreter runs the code without error.

`__qualname__` has [been available since Python 3.3](https://docs.python.org/3/library/stdtypes.html#definition.__qualname__).

Tested on _Python 3.7.1_, using _mpypy 0.670_.

**Comment by ilevkivskyi** [comment]
Such kind of issues are typically easy to fix. You can look at how `implicit_module_attrs()` are implemented, and do the same for classes. Note however that we are currently migrating to a new semantic analyzer, so the fix should apply to both.

**Comment by jtmoon79** [comment]
Thanks @ilevkivskyi .  I'm curious, why was this Issue labeled "false-positive"?

**Comment by gvanrossum** [comment]
The label "false-positive" refers to the fact that mypy gives an error where it shouldn't. It doesn't mean there's anything wrong with your bug report!

**Comment by aaalex5** [comment]
If no one is working on this issue, I'd like to give it a shot if possible. :)

**Comment by ilevkivskyi** [comment]
Go ahead! But please be aware that we are currently working on big refactoring on semantic analyzer, so you will need to copy your fix to the `newsemanal` as well.

**Comment by aaalex5** [comment]
Just making sure that the modification should be made in the nodes.py file under the following:
```python
implicit_module_attrs = {'__name__': '__builtins__.str',
                         '__doc__': None,  # depends on Python version, see semanal.py
                         '__file__': '__builtins__.str',
                         '__package__': '__builtins__.str'}  # type: Final
```

**Comment by aaalex5** [comment]
So I believe I fixed the issue within the normal mypy directory, but I'm not sure where in newsemanal it would be necessary to make a change? The only place I can think to make additions or changes would be in newsemanal/semanal_pass1.py, otherwise it doesn't look like there would be a change needed other than the one in mypy/mypy/nodes.py where ```implicit_module_attrs()``` is

**Comment by ilevkivskyi** [comment]
I don't think using/changing `implicit_module_attrs` is the best solution. `__qualname__` is only defined at class scope, at module scope it is a `NameError`. Probably we should have something like `implicit_class_attrs`.

**Comment by aaalex5** [comment]
Any tips on implementing that? implicit_module_attrs uses a set with 'module name attributes' keys and 'type' values so for an implicit_class_attrs would there be even be a 'type'? I'm just wondering if a set would even be the correct solution. Also, what other classes, if any, would you recommend I add?

**Comment by forrestchang** [comment]
@aaalex5 I started a pull request https://github.com/python/mypy/pull/7683.

**Comment by asottile** [comment]
#10570 is a duplicate with slightly more details
